mslib/msui/mpl_qtwidget.py

```diff
@@ -327,25 +320,24 @@ def __init__(self, fig=None, ax=None, settings=None, numlabels=None, num_interpo
         super().__init__(fig, ax)
 
         # Default settings.
-        self.settings_dict = {"vertical_extent": (1050, 180),
-                              "vertical_axis": "pressure",
-                              "secondary_axis": "no secondary axis",
-                              "plot_title_size": "default",
-                              "axes_label_size": "default",
-                              "flightlevels": [],
-                              "draw_flightlevels": True,
-                              "draw_flighttrack": True,
-                              "fill_flighttrack": True,
-                              "label_flighttrack": True,
-                              "draw_verticals": True,
-                              "draw_marker": True,
-                              "draw_ceiling": True,
-                              "colour_ft_vertices": (0, 0, 1, 1),
-                              "colour_ft_waypoints": (1, 0, 0, 1),
-                              "colour_ft_fill": (0, 0, 1, 0.15),
-                              "colour_ceiling": (0, 0, 1, 0.15)}
-        if settings is not None:
-            self.settings_dict.update(settings)
+        self.set_settings({"vertical_extent": (1050, 180),
+                           "vertical_axis": "pressure",
+                           "secondary_axis": "no secondary axis",
+                           "plot_title_size": "default",
+                           "axes_label_size": "default",
+                           "flightlevels": [],
+                           "draw_flightlevels": True,
+                           "draw_flighttrack": True,
+                           "fill_flighttrack": True,
+                           "label_flighttrack": True,
+                           "draw_verticals": True,
+                           "draw_marker": True,
+                           "draw_ceiling": True,
+                           "colour_ft_vertices": (0, 0, 1, 1),
+                           "colour_ft_waypoints": (1, 0, 0, 1),
+                           "colour_ft_fill": (0, 0, 1, 0.15),
+                           "colour_ceiling": (0, 0, 1, 0.15)})
+        self.set_settings(settings)
```

Comment:
"draw_graticule" is not defined here, but looks like mandatory

Reply:
only for TopView. These settings are heavily related to the Options in the UI. One can think about a better way of handling this without welding this code again with QT.